machines/store.ts

```diff
@@ -325,15 +354,24 @@ export async function getItem(
   encryptionKey: string
 ) {
   try {
-    const data = await Storage.getItem(key);
+    const data = await Storage.getItem(key, encryptionKey);
     if (data != null) {
       const decryptedData = decryptJson(encryptionKey, data);
       return JSON.parse(decryptedData);
+    }
+    if (data === null && vcKeyRegExp.exec(key)) {
+      await Storage.removeItem(key);
+      removeVCMetaData(MY_VCS_STORE_KEY, key, encryptionKey);
+      throw new Error('Data is tampered');
     } else {
       return defaultValue;
     }
   } catch (e) {
-    return defaultValue;
+    if (e.message.includes('Data is tampered')) {
+      throw e;
```

Comment:
If throw stops the flow of execution can we modify this?
`if (e.message.includes('Data is tampered')) {
      throw e;
return defaultValue;`